impeller/typographer/backends/skia/text_render_context_skia.cc

```diff
@@ -340,25 +340,37 @@ static std::shared_ptr<Texture> UploadGlyphTextureAtlas(
 
 std::shared_ptr<GlyphAtlas> TextRenderContextSkia::CreateGlyphAtlas(
     GlyphAtlas::Type type,
+    std::shared_ptr<GlyphAtlasContext> atlas_context,
     FrameIterator frame_iterator) const {
   TRACE_EVENT0("impeller", __FUNCTION__);
   if (!IsValid()) {
     return nullptr;
   }
-
-  auto glyph_atlas = std::make_shared<GlyphAtlas>(type);
+  auto last_atlas = atlas_context->GetGlyphAtlas();
 
   // ---------------------------------------------------------------------------
   // Step 1: Collect unique font-glyph pairs in the frame.
   // ---------------------------------------------------------------------------
 
   auto font_glyph_pairs = CollectUniqueFontGlyphPairs(type, frame_iterator);
   if (font_glyph_pairs.empty()) {
-    return glyph_atlas;
+    return last_atlas;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Step 2: Determine if the atlas type and font glyph pairs are compatible
+  //         with the current atlas and reuse if possible.
+  // ---------------------------------------------------------------------------
+  if (last_atlas->GetType() == type &&
+      last_atlas->HasSamePairs(font_glyph_pairs)) {
+    return last_atlas;
```

Comment:
We should file an issue to explore copying out the glyphs we still need. Or maybe just copy the whole atlas if it's small enough.

Reply:
I was thinking something like, if you need a small number of additional glyphs and they fit on the current atlas, just append them and re-upload. Even if we resized the atlas, reusing and blitting the old atlas instead of each character would be a good idea too

Reply:
We'd either have to keep the rect packer around or store the  taken/free area somehow.

Reply:
Yeah, this probably deserves a doc exploring the options we have here

Reply:
I explored this a bit while I was out. There are a lot of potential caching options available to us, but I think we should probably keep pulling on making the whole thing faster first before we invest too much effort into tuning this.

To that end, lets land the trivial case caching and circle back?

Reply:
SGTM